### PR TITLE
Search router

### PR DESCRIPTION
**Pull request by allyjay317**
---------------------------
[Search Route](https://trello.com/c/LlSRCnFc/78-search-route)
---------------------------
## Changed
 - profiles.test.js
 - searchRouter.js
 - package-lock.json
 - package.json
## New
 - searchModel.js
## Features
 - new search functionality
   - will now accept a title, category name, zip code and address in a query to filter search results
   - if no zip or address is given, will default to profile's address for sorting
   - gives distance to item in meters
## Issues
 - no issue currently, but with a larger item amount, pagination of results would be best

### CODE REVIEW

**Review by chiku524 — APPROVED** [review]

**Review by PjDeVo — APPROVED** [review]

File: api/search/searchRouter.js [hunk 1]
```diff
@@ -1,17 +1,107 @@
...
+      );
+    }
+    res.send(items);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({ message: 'a server error occured' });
+  }
+});
 
 router.get('/testdata', async (req, res) => {
```
Comment (tinomen) [inline api/search/searchRouter.js:60]: If this was added so the front end can move forward before search was done then lets remove it.

File: api/search/searchRouter.js [hunk 2]
```diff
@@ -1,17 +1,107 @@
...
+};
+
+const distanceSearch = async (items, zip, address) => {
+  const query = zip ? zip : address;
+  const geoRes = await geocodeService
+    .forwardGeocode({ query, limit: 1 })
+    .send();
+  const zipLat = geoRes.body.features[0].center[0];
+  const zipLng = geoRes.body.features[0].center[1];
+  items = await Promise.all(
```
Comment (tinomen) [inline api/search/searchRouter.js:86]: This loop could be broken out into a nice simple function

File: api/search/searchRouter.js [hunk 3]
```diff
@@ -1,17 +1,107 @@
...
+    items.map(async (i) => {
+      i.distance = await getDistance(
+        { latitude: zipLat, longitude: zipLng },
+        { latitude: i.latitude, longitude: i.longitude }
+      );
+      return i;
+    })
+  );
+  items = items.sort((a, b) => {
```
Comment (tinomen) [inline api/search/searchRouter.js:95]: another good candidate for a small function

**Comment by tinomen** [comment]
You can ignore my comments, just doing a critique with the TPLs.

BTW very nice PR description and commits. It's a nice sized PR and easy to follow.

**Pull request merged**